enterprise/frontend/src/embedding-sdk/sdk-package/components/public/MetabaseProvider/MetabaseProvider.tsx

```diff
@@ -0,0 +1,114 @@
+import { memo, useMemo } from "react";
+import useDeepCompareEffect from "react-use/lib/useDeepCompareEffect";
+
+import { ClientSideOnlyWrapper } from "embedding-sdk/sdk-package/components/private/ClientSideOnlyWrapper/ClientSideOnlyWrapper";
+import { useInitializeMetabaseProviderPropsStore } from "embedding-sdk/sdk-package/hooks/private/use-initialize-metabase-provider-props-store";
+import { useLoadSdkBundle } from "embedding-sdk/sdk-package/hooks/private/use-load-sdk-bundle";
+import { RenderSingleCopy } from "embedding-sdk/sdk-shared/components/RenderSingleCopy/RenderSingleCopy";
+import { useMetabaseProviderPropsStore } from "embedding-sdk/sdk-shared/hooks/use-metabase-provider-props-store";
+import { useSdkLoadingState } from "embedding-sdk/sdk-shared/hooks/use-sdk-loading-state";
+import { ensureMetabaseProviderPropsStore } from "embedding-sdk/sdk-shared/lib/ensure-metabase-provider-props-store";
+import { getWindow } from "embedding-sdk/sdk-shared/lib/get-window";
+import type { SdkStore } from "embedding-sdk/store/types";
+import type { MetabaseProviderProps } from "embedding-sdk/types/metabase-provider";
+
+type MetabaseProviderInitDataWrapperProps = Pick<
+  MetabaseProviderProps,
+  "authConfig" | "allowConsoleLog"
+> & {
+  reduxStore: SdkStore;
+};
+
+/**
+ * We call `use-init-data` hook to initialize the SDK with the initial data.
+ * This is necessary when hooks are used before any SDK component is rendered.
+ */
+const MetabaseProviderInitDataWrapper = memo(function InitDataWrapper({
+  authConfig,
+  allowConsoleLog,
+  reduxStore,
+}: MetabaseProviderInitDataWrapperProps) {
+  const useInitData = window?.MetabaseEmbeddingSDK?.useInitData;
+
+  if (!reduxStore || !useInitData) {
+    throw new Error('Embedding SDK "useInitData" hook is not available');
+  }
+
+  useInitData({
+    reduxStore,
+    authConfig,
+    allowConsoleLog,
+  });
+
+  return null;
+});
+
+const MetabaseProviderInner = memo(function MetabaseProviderInner(
+  props: Omit<MetabaseProviderProps, "children">,
+) {
+  useLoadSdkBundle(props.authConfig.metabaseInstanceUrl);
+
+  const { isLoading } = useSdkLoadingState();
+
+  const {
+    props: { reduxStore: existingStore },
+  } = useMetabaseProviderPropsStore();
+
+  // Return existing store or create a new one
+  const reduxStore = useMemo(
+    () =>
+      isLoading
+        ? null
+        : (existingStore ?? getWindow()?.MetabaseEmbeddingSDK?.getSdkStore()),
```

Comment:
`existingStore` is set after the MetabaseProvider is remounted, or when there are multiple providers